### PR TITLE
[1227] Fixed Eslint warnings

### LINKED ISSUE #1227: Fix eslint warnings spamming our PRs
Our lovely GitHub Actions code is adding inline eslint warnings to our PRs:

Screen Shot 2020-10-20 at 9 55 10 PM

Let's fix the warnings so we don't have these.

### PR DESCRIPTION
**Pull request by birtony**
## Issue This PR Addresses
Fixes #1227 

## Type of Change

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

@ceterisp has originally worked on issue #1227 in https://github.com/Seneca-CDOT/telescope/pull/1391. I have looked at the warnings Eslint used to throw and there wasn't anything related to https://github.com/Seneca-CDOT/telescope/pull/1391/files#diff-53149432e2cca92090a6cb3603c9570f20ebf6546280484ece66ab937794283bR79. All of the warnings should be addressed in this pr. Therefore, when merged, I'd suggest closing https://github.com/Seneca-CDOT/telescope/pull/1391

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/frontend/gatsby/src/components/Post/Post.jsx [hunk 1]
```diff
@@ -101,7 +101,11 @@ const Post = ({ postUrl }) => {
   // We need a ref to our post content, which we inject into a <section> below.
   const sectionEl = useRef(null);
   // Grab the post data from our backend so we can render it
-  const { data: post, error } = useSWR(postUrl, (url) => fetch(url).then((r) => r.json()));
+  const { data: post, error } = useSWR(postUrl, async (url) => {
```
Comment (humphd) [inline src/frontend/gatsby/src/components/Post/Post.jsx:104]: You can simplify this, and use `useSWR(postUrl)` with no second arg (the default fetcher will get used).

File: src/frontend/gatsby/src/components/Posts/Posts.jsx [hunk 2]
```diff
@@ -37,7 +37,10 @@ const Posts = () => {
   const [currentPostId, setCurrentPostId] = useState(null);
   const { data, size, setSize, error } = useSWRInfinite(
     (index) => `${telescopeUrl}/posts?page=${index + 1}`,
-    (url) => fetch(url).then((r) => r.json()),
+    async (url) => {
```
Comment (humphd) [inline src/frontend/gatsby/src/components/Posts/Posts.jsx:40]: Same deal here, we can drop this function completely.

**Review by humphd — CHANGES_REQUESTED** [review]

**Review by humphd — APPROVED** [review]

File: src/frontend/gatsby/src/components/Post/Post.jsx [hunk 3]
```diff
@@ -101,7 +101,8 @@ const Post = ({ postUrl }) => {
   // We need a ref to our post content, which we inject into a <section> below.
   const sectionEl = useRef(null);
```
Comment (HyperTHD) [inline src/frontend/gatsby/src/components/Post/Post.jsx:102]: Possible type checking here with the useRef?

**Comment by humphd** [comment]
I am so happy to have these lint issues cleaned up!

**Pull request merged**